src/MongoDB.Bson/ObjectModel/BsonObjectId.cs

```diff
@@ -71,8 +71,10 @@ public BsonObjectId(DateTime timestamp, int machine, short pid, int increment)
         /// <param name="machine">The machine hash.</param>
         /// <param name="pid">The PID.</param>
         /// <param name="increment">The increment.</param>
-        [Obsolete("Use new BsonObjectId(new ObjectId(int timestamp, int machine, short pid, int increment)) instead.")]
-        public BsonObjectId(int timestamp, int machine, short pid, int increment)
+        [Obsolete("Use new BsonObjectId(new ObjectId(uint timestamp, int machine, short pid, int increment)) instead.")]
+#pragma warning disable CS3001
+        public BsonObjectId(uint timestamp, int machine, short pid, int increment)
+#pragma warning restore CS3001
```

Comment:
I suggest that this file need not be changed at all. See comments in ObjectId.cs also.